### ISSUE TITLE
Error of size 

### ISSUE THREAD

**Issue by ymmm-4** [YUCOCOCS/Swin-TransUper#1]
Where to set it within the code ? 
    assert inputs.size() == target.size(), 'predict {} & target {} shape do not match'.format(inputs.size(), target.size())
AssertionError: predict torch.Size([8, 1, 224, 224]) & target torch.Size([8, 2, 224, 224]) shape do not match

**Comment by YUCOCOCS** [comment]
Can you explain the details of your training? It seems like there was a category setting error

**Comment by ymmm-4** [comment]
My images are grayscale. I ran swin-unet on that and it worked so I am using same on this code

**Comment by YUCOCOCS** [comment]
What dataset did you train on, ISIC2017?

**Comment by ymmm-4** [comment]
Retinal images

**Comment by YUCOCOCS** [comment]
If you are a binary dataset, use the training settings and code of ISIC2017.

**Comment by ymmm-4** [comment]
yes it is binary. do I need to change low, middle or high thing?

**Comment by YUCOCOCS** [comment]
Set the number of categories to 1 and use torch. nn BCEWithLogitsLoss should be used for loss, and your label should be single channel.

**Comment by ymmm-4** [comment]
what to add in swin_transuper.py? my training data is according to binary but this code is for different channels

**Comment by YUCOCOCS** [comment]
I don't know what the details of your code are, this error represents a different number of channels when calculating the loss. Please carefully examine the changes in the channel dimension of the code

**Comment by ymmm-4** [comment]
I think we can do it by adding something in your code because my train n test code is already for binary, I ran swinunet code as well on that

**Comment by ymmm-4** [comment]
May be the pre-trained are for RGB

**Comment by ymmm-4** [comment]
how to modify pre-train
